### PR TITLE
feat: optional `push` and `commit` input

### PR DESCRIPTION
**Pull request by nbrugger-tgm**
Implementation of https://github.com/commitizen-tools/commitizen-action/issues/15

### CODE REVIEW

**Comment by nbrugger-tgm** [comment]
Passing test on : https://github.com/nbrugger-tgm/commitizen-action/pull/1

**Comment by Woile** [comment]
Awesome work!

**Comment by nbrugger-tgm** [comment]
Had to re-commit since tests didn't work with a fork->upstream PR

File: .github/workflows/test_action.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      - opened
+      - synchronize
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.head_ref }}
```
Comment (Woile) [inline .github/workflows/test_action.yml:13]: Can you try `base_ref` here? thanks!

Comment (nbrugger-tgm) [reply .github/workflows/test_action.yml:13]: Well it wouldnt test my code i think because the push would happen to head i think

Comment (nbrugger-tgm) [reply .github/workflows/test_action.yml:13]:
```yaml
ref: ${{github.event.pull_request.head.ref}}
repository: ${{github.event.pull_request.head.repo.full_name}}
```

found this from a gitub discussion (posted by staff)

**Comment by nbrugger-tgm** [comment]
Sorry made a mistake, tested in on my repo and it worked.

I am sorry about the inconvenience but testing workflows especialy with forks etc is a little pain

**Comment by Woile** [comment]
No problem, thanks a lot actually for taking the time to do this. Can I ask you to squash into a single commit?
Thanks!

**Comment by Woile** [comment]
Thanks for your contribution!

**Pull request merged**